internal/nodes/crd_translator.go

```diff
@@ -55,6 +55,11 @@ func BuildL7HostRule(host, key string, vsNode AviVsEvhSniModel) {
 		} else if hostrule.Status.Status == lib.StatusRejected {
 			// do not apply a rejected hostrule, this way the VS would retain
 			return
+		} else {
+			if lib.GetTenantInNamespace(hostrule.Namespace) != vsNode.GetTenant() {
+				utils.AviLog.Warnf("key: %s, msg: Tenant annotation in hostrule namespace %s does not matches with the tenant of host %s ", key, hostrule.Namespace, host)
```

Comment:
Shouldn't the CRD be rejected here already? Don't think we need this check IMO

Reply:
When hostrule have a hostname sharedVS related setting, then at that time this check will be required.